### PR TITLE
Improve File Import: Error Handling for Zip Validation 

### LINKED ISSUE #7050: Improve logs for "This isn’t a valid Anki package file"
User report: AnKing v7 fails to import

No exception report is generated. (2.13 beta 1 and 2.12.1)

`    <string name="import_log_no_apkg">This isn’t a valid Anki package file</string>`

We need to determine which path this takes, and ideally add a (possibly silent) exception report on this path,

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
A user didn't understand "archive is not a ZIP archive" meant the file is corrupt. We improve this error message, and perform more silent exception logging to help diagnose these unknown issues in the future.

## Fixes
Fixes #7050 

## Approach
* Change the dialog
* Send a silent exception report if the exception type is unknown
* Return a class so we can better handle displaying a string to the user.

## How Has This Been Tested?

image

## Learning (optional, can help others)
We can always do better with exception reports

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java [hunk 1]
```diff
@@ -151,51 +155,51 @@ private String handleContentProviderFile(Context context, Intent intent, Uri dat
...
                 sendShowImportFileDialogMsg(tempOutDir);
-                return null;
+                return ImportResult.fromSuccess();
             }
         }
 
-
-        protected String validateZipFile(Context context, String filePath) {
+        @NonNull
+        protected ImportResult validateZipFile(Context ctx, String filePath) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java:195]: I was ambivalent on this - but it's better as an Optional, I'll fix

**Review by mikehardy — APPROVED** [review]
This should be a help - always a tough moment for users and the more info we can give them to self-help the better

**Comment by mikehardy** [comment]
Marked for a pick to release-2.13 too - we haven't skewed at all yet no reason this can't go in

**Comment by mikehardy** [comment]
This one and the other messaging one have been language synced, then picked (along with language sync) to release-2.13, 2.13.0beta2 coming up shortly

**Pull request merged**